lib/sinon/sandbox.js

```diff
@@ -349,7 +349,8 @@ function Sandbox() {
 
     sandbox.useFakeXMLHttpRequest = function useFakeXMLHttpRequest() {
         var xhr = fakeXhr.useFakeXMLHttpRequest();
-        return push.call(collection, xhr);
+        push.call(collection, xhr);
+        return xhr;
```

Comment:
Why was this change made? There should be a test to confirm this change in functionality.

Reply:
This is to keep behaviour of `sinon.useFakeXMLHttpRequest` the same. 

Sinon object has been made the default sandbox but it doesn't use `sandbox.useFakeXMLHttpRequest` method, making it impossible to restore fake xhr object using `sinon.restore`.

This PR changes it so sinon object uses `sandbox.useFakeXMLHttpRequest`. The behaviour of the latter was, however, slightly different as it would return: 

`return push.call(...)`

It means it returns *the length of the internal fake collection* (what doesn't make any sense) instead of fake xhr object returned by nise.

I've added a test to sandbox-test.js.